vms/platformvm/txs/base_tx_test.go

```diff
@@ -10,65 +10,443 @@ import (
+		0xff, 0xee, 0xdd, 0xcc, 0xbb, 0xaa, 0x99, 0x88,
+		// Tx output index
+		0x00, 0x00, 0x00, 0x01,
+		// Mainnet AVAX assetID
+		0x21, 0xe6, 0x73, 0x17, 0xcb, 0xc4, 0xbe, 0x2a,
+		0xeb, 0x00, 0x67, 0x7a, 0xd6, 0x46, 0x27, 0x78,
+		0xa8, 0xf5, 0x22, 0x74, 0xb9, 0xd6, 0x05, 0xdf,
+		0x25, 0x91, 0xb2, 0x30, 0x27, 0xa8, 0x7d, 0xff,
+		// secp256k1fx transfer input type ID
+		0x00, 0x00, 0x00, 0x05,
+		// input amount = 1 MilliAvax
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x0f, 0x42, 0x40,
+		// number of signatures needed in input
+		0x00, 0x00, 0x00, 0x01,
+		// index of signer
+		0x00, 0x00, 0x00, 0x05,
+		// length of memo
+		0x00, 0x00, 0x00, 0x00,
+	}
+	var unsignedSimpleBaseTx UnsignedTx = simpleBaseTx
+	unsignedSimpleBaseTxBytes, err := Codec.Marshal(Version, &unsignedSimpleBaseTx)
+	require.NoError(err)
+	require.Equal(expectedUnsignedSimpleBaseTxBytes, unsignedSimpleBaseTxBytes)
+
+	complexBaseTx := &BaseTx{
+		BaseTx: avax.BaseTx{
+			NetworkID:    constants.MainnetID,
+			BlockchainID: constants.PlatformChainID,
+			Outs: []*avax.TransferableOutput{
+				{
+					Asset: avax.Asset{
+						ID: avaxAssetID,
+					},
+					Out: &stakeable.LockOut{
+						Locktime: 87654321,
+						TransferableOut: &secp256k1fx.TransferOutput{
+							Amt: 1,
+							OutputOwners: secp256k1fx.OutputOwners{
+								Locktime:  12345678,
+								Threshold: 0,
+								Addrs:     []ids.ShortID{},
+							},
+						},
+					},
+				},
+				{
+					Asset: avax.Asset{
+						ID: customAssetID,
+					},
+					Out: &stakeable.LockOut{
+						Locktime: 876543210,
+						TransferableOut: &secp256k1fx.TransferOutput{
+							Amt: 0xffffffffffffffff,
+							OutputOwners: secp256k1fx.OutputOwners{
+								Locktime:  0,
+								Threshold: 1,
+								Addrs: []ids.ShortID{
+									addr,
+								},
+							},
+						},
+					},
+				},
+			},
+			Ins: []*avax.TransferableInput{
+				{
+					UTXOID: avax.UTXOID{
+						TxID:        txID,
+						OutputIndex: 1,
+					},
+					Asset: avax.Asset{
+						ID: avaxAssetID,
+					},
+					In: &secp256k1fx.TransferInput{
+						Amt: units.Avax,
+						Input: secp256k1fx.Input{
+							SigIndices: []uint32{2, 5},
+						},
+					},
+				},
+				{
+					UTXOID: avax.UTXOID{
+						TxID:        txID,
+						OutputIndex: 2,
+					},
+					Asset: avax.Asset{
+						ID: customAssetID,
+					},
+					In: &stakeable.LockIn{
+						Locktime: 876543210,
+						TransferableIn: &secp256k1fx.TransferInput{
+							Amt: 0xefffffffffffffff,
+							Input: secp256k1fx.Input{
+								SigIndices: []uint32{0},
+							},
+						},
+					},
+				},
+				{
+					UTXOID: avax.UTXOID{
+						TxID:        txID,
+						OutputIndex: 3,
+					},
+					Asset: avax.Asset{
+						ID: customAssetID,
+					},
+					In: &secp256k1fx.TransferInput{
+						Amt: 0x1000000000000000,
+						Input: secp256k1fx.Input{
+							SigIndices: []uint32{},
+						},
+					},
+				},
 			},
+			Memo: types.JSONByteSlice("😅\nwell that's\x01\x23\x45!"),
```

Comment:
Why is there an emoji?

Reply:
This is just to make sure that non-ascii strings are handled well